content/blog/posts/2021-03-11-getting-started-with-serverless.md

```diff
@@ -0,0 +1,174 @@
+---
+authorId: chuckcarpenter
+categories:
+  - graphql
+  - javascript
+  - serverless
+date: '2021-03-11'
+slug: getting-started-with-serverless
+title: Getting Started with Serverless
+---
+
+While Ship Shape engineers are deep practioners of JavaScript and are often working on web applications for our clients, we sometimes are also creating services for these applications to service functions for the app or provide the data layer and persistance in the form of an API.
+
+We've used a number of languages for these services, but usually default to our expertise in JavaScript and work within the Node.js run-time. This works well and we then integrate into whatever deployment pipeline the client may have in place. Recently, we've started creating these services with a serverless cloud solution, rather than deploy to server infrastructure.
+
+Serverless allows abstraction from the infrastructure. You will not have to provision, scale, or maintain servers to execute your applications, databases and/or storage systems. There is no active management, so developers are more empowered to create functions as a service and deploy directly, as well as frequently.
+
+## What we'll work on today
+
+A common need for teams today in their applications is a GraphQL server as the API for data persistance. While not challenging to setup a basic service for use locally, there tend to often be quite a bit more steps for deployment of these APIs for use in various environments. We'll use the Serverless framework to get us started quickly and give us the ability to deploy to many common cloud providers. 
+
+```bash
+npx serverless
+```
+
+Follow the prompts and this will give you a basic serverless project with all you need to get started. We can change the `serverless.yml` to define our function and the handler specifically for HTTP requests by adding the following:
```

Comment:
```suggestion
If we follow the prompts, this will give us a basic serverless project with all we need to get started. We can change the `serverless.yml` to define our function and the handler specifically for HTTP requests by adding the following:
```